src/drawGenerator.php

```diff
@@ -0,0 +1,154 @@
+<?php
+
+  require_once "src/abstract.php";
+  require_once "src/interface.php";
+
+class Square extends Shape implements IShapeDrawSquare, IShapeSetColor
+{
+  protected $length = 0;
+
+  private $colors = array("red" => 0, "green" => 0, "blue" => 0);
+
+  private $isRandom = false;
+
+  public function show() {
+    if(!($this->checkSidesValue($this->length))) 
+      return false;
+
+    $this->draw();
+  }
+  
+  public function getShapeName() : string {
+    return $this->shapeName;
+  }
+
+  public function draw() {
+    for($i = 0; $i < $this->length; ++$i) {
+      for($k = 0; $k < $this->length; ++$k) {
+        echo "<div class='block' style='float: left;
+        height: ".Shape::DRAWING_SCALE_X."px; width: ".Shape::DRAWING_SCALE_Y."px; background-color: ".$this->getRGBtoHex().";'></div>";
+      }
+      echo "<div style='clear:both;'></div><br>";
+    }
+  }
+
+  public function checkSidesValue($length) {
```

Comment:
I do not understand what this function exactly does. Please rename to checkIfSides....... for exaple checkIfSidesMoreThanZero. 

Function name is in plural form `checkSides` but it takes only one parameter length. Consider rename to checkIfLength.... or checkIfSide...

Reply:
you're right